internal/datastore/data_store_status_provider_impl.go

```diff
@@ -2,19 +2,33 @@ package datastore
 
 import (
 	"github.com/launchdarkly/go-server-sdk/v7/interfaces"
-	"github.com/launchdarkly/go-server-sdk/v7/subsystems"
 )
 
+type StatusMonitorable interface {
```

Comment:
This refactoring defines an interface with a single method - one of DataStore's methods. This component only needed that single method; this interface documents that fact. 

It also makes it _way easier_ to pass in something to satisfy it, instead of having to implement a bunch of no-op methods.